server/controllers/authControllers.ts

```diff
@@ -56,6 +57,64 @@ export const registerController = async (req: Request, res: Response): Promise<v
 
     }
 }
+export const editController = async (req: Request, res: Response): Promise<void | void> => {
+    try {
+        const toEdit: { [x: string]: string; password: string; _id: string } = req.body;
+        console.log('to edit: ', toEdit)
+        const fieldToUpdate = Object.keys(toEdit).find(key => key !== 'password' && key !== '_id');
+        if (!fieldToUpdate) {
+            res.status(400);
+            throw new Error('could not read field to update')
+        }
+        const filter = { _id: toEdit._id };
+        const potentialUser = await UserModel.findOne(filter);
+        if (!potentialUser) {
+            res.status(400);
+            throw new Error('no user found')
+        }
+
+        const isMatch = await bcrypt.compare(toEdit.password, potentialUser.password);
```

Comment:
Hi Jonas -- yes, that is a good point! I will remove it here to simplify

Reply:
Sooo the reason i've done this is to add an extra security layer when editing -- if it's too much I can make this effective only if the field user is editing the password (this would only prompt the password when entering password). let me know what you think.

Reply:
No, it makes sense as a security measure and I have seen this behavior elsewhere. I just wasn't sure if that's the purpose. Also, I'm wondering if you would usually solve this by calling the "normal" auth endpoint right before sending the data. But this works and I'd vote to leave it like this for now. We can always ask Vic for her opinion on implementation specifics tomorrow...